### PR TITLE
Fixes failing unit tests on travis

### LINKED ISSUE #8720: PHP 5.5 and 5.6 builds failing on travis (as of April 27, 2016)
This is a (multiple allowed):
* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 3.2.8
* Platform and Target: PHP 5.5.21, PhpUnit 4.8.24, MySql
* Platform and Target: PHP 5.6.5, PhpUnit 5.3.2, MySql

### What you did

Created a couple of PRs and noticed the builds on travis for the above platforms are failing.

### Expected Behavior

The following tests should pass.

```
 Cake\Test\TestCase\Database\QueryTest::testSelectWhereOperatorLessThan
 Cake\Test\TestCase\Database\QueryTest::testSelectWithJoinsConditions
```

### Actual Behavior

I'm going to configure a machine with PHP 5.5 and see if I can reproduce the problem, and make a PR to fix the issue.

### PR DESCRIPTION
**Pull request by thinkingmedia**
This PR fixes #8720 

Travis YAML has been updated to run Ubuntu 14.04 and MySQL 5.6.

This PR has all test passing without modifying any of the tests (based upon my upstream from yesterday).

Unit test failures were being caused by MySQL truncating fixture tables. The issue appears to be random but consistent when it happens. The work around was to load fixtures manually in the unit test.

I'm happy with this fix as it more closely matches production environments for CakePHP apps. I don't think the majority of people are going to be using Ubuntu 12 and MySQL 5.6 is also very common. I don't think there are any breaking changes between 5.5 and 5.6.

### CODE REVIEW

**Comment by thinkingmedia** [comment]
Looks like this is not ready to be merged.

File: .travis.yml [hunk 1]
```diff
@@ -5,11 +5,18 @@ php:
   - 5.6
   - 7.0
 
-sudo: false
+dist: trusty
+sudo: required
```
Comment (markstory) [inline .travis.yml]: Neat. We can give the [new VM infrastructure](https://docs.travis-ci.com/user/trusty-ci-environment/) a try.

**Comment by markstory** [comment]
If you can squash up the noisy commits, I can get this merged in.

**Comment by ADmad** [comment]
`hhvm` is not available through `phpenv` when using `trusty` distro on travis:  https://travis-ci.org/cakephp/cakephp/jobs/126814574

**Comment by thinkingmedia** [comment]
I was so focused on fixing PHP 5.5 that I broke all the other environments. At least I'm learning a lot about how Travis works. I don't know what to do now. I do not know if you can upgrade MySQL to 5.6 on the Ubuntu 12 image.

i'm open to ideas.

**Comment by ADmad** [comment]
You didn't break anything :)  It's just that `hhvm` jobs don't run since it's not available on travis' `trusty` VMs. Their [doc](https://docs.travis-ci.com/user/trusty-ci-environment/#Environment-common-to-all-VM-images) says having various PHP versions pre installed in the trusty VM is at the top of list for their next updates. So let's wait and see. Currently all builds are passing with recent changes done to fixture loading for tests. So we don't really need trusty for now.

**Comment by markstory** [comment]
We might as well remove the HHVM builds. They basically never pass as HHVM has a few incompatibilities around datetimes and intl.

**Comment by ADmad** [comment]
@thinkingmedia Can you please remove hhvm from the matrix and squash the commits.

**Comment by thinkingmedia** [comment]
Sorry guys. I had a rough Friday. You still want me to change this?

**Comment by ADmad** [comment]
@thinkingmedia Nope, I took care of it.

**Comment by thinkingmedia** [comment]
@ADmad thank you.

**Pull request merged**